src/components/comment-item.tsx

```diff
@@ -66,7 +66,7 @@ export const CommentItem: React.VoidFunctionComponent<CommentItemProps> = ({
         <div>
           <CommentAuthor>{comment.authorName}</CommentAuthor>:{" "}
           <CommentDate>{formatDate(comment.createdAt)}</CommentDate>
-          {comment.isPublished ? undefined : (
+          {comment.status === "pending" ? undefined : (
```

Comment:
Here you can use const with map enum

Reply:
Using enum-like consts is a common pattern in JS and some other languages. We don’t need this extra abstraction in TypeScript as it does not add extra value. Both autocompletion and linting are available for string literals:

<img width="716" alt="Screenshot 2022-02-06 at 17 05 02" src="https://user-images.githubusercontent.com/608862/152692163-4a30ad18-be75-4b94-b9fe-17f34fcbbd17.png">

<img width="962" alt="Screenshot 2022-02-06 at 17 05 20" src="https://user-images.githubusercontent.com/608862/152692165-a46042dc-a7a8-4b42-8287-07d91f4d6eca.png">

Maintaining an extra const instead means that we have to import it from somewhere. This may lead to circular deps, which are hard to untangle.

TypeScript docs: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions